tests/test_endpoints.py

```diff
@@ -27,6 +27,15 @@ class EndpointFromEnvironment:
     search = "search"
 
 
+class BaseEndpoint:
```

Comment:
```suggestion
@endpoint(base_url="https://food.com")
class BaseEndpoint:
```

Reply:
There is a problem when you do that, since the decorator is equivalent to `BaseEndpoint = endpoint(BaseEndpoint)` all attributes from `BaseEndpoint` have been modified, so when `SubClass` sets it's base_url, it ends with something like this `https://fruits.com/https://food.com/apples`. I'll be working on a way to avoid this behavior.

Reply:
I haven't found a way to solve this without needing to store the base_url inside the class